lib/features/composer/domain/usecases/create_new_and_save_email_to_drafts_interactor.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:jmap_dart_client/jmap/core/session/session.dart';
+import 'package:jmap_dart_client/jmap/core/state.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email.dart';
+import 'package:tmail_ui_user/features/composer/domain/exceptions/compose_email_exception.dart';
+import 'package:tmail_ui_user/features/composer/domain/repository/composer_repository.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/generate_email_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/save_email_as_drafts_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/update_email_drafts_state.dart';
+import 'package:tmail_ui_user/features/composer/presentation/model/create_email_request.dart';
+import 'package:tmail_ui_user/features/email/domain/exceptions/email_exceptions.dart';
+import 'package:tmail_ui_user/features/email/domain/repository/email_repository.dart';
+import 'package:tmail_ui_user/features/mailbox/domain/repository/mailbox_repository.dart';
+import 'package:tmail_ui_user/main/exceptions/remote_exception.dart';
+
+class CreateNewAndSaveEmailToDraftsInteractor {
+  final EmailRepository _emailRepository;
+  final MailboxRepository _mailboxRepository;
+  final ComposerRepository _composerRepository;
+
+  CreateNewAndSaveEmailToDraftsInteractor(
+    this._emailRepository,
+    this._mailboxRepository,
+    this._composerRepository,
+  );
+
+  Stream<dartz.Either<Failure, Success>> execute({
+    required CreateEmailRequest createEmailRequest,
+    CancelToken? cancelToken,
+  }) async* {
+    try {
+      yield dartz.Right<Failure, Success>(GenerateEmailLoading());
+
+      final listCurrentState = await _getStoredCurrentState(
+        session: createEmailRequest.session,
+        accountId: createEmailRequest.accountId
+      );
+
+      final emailCreated = await _createEmailObject(createEmailRequest);
+
+      if (emailCreated != null) {
+        if (createEmailRequest.draftsEmailId == null) {
+          yield dartz.Right<Failure, Success>(SaveEmailAsDraftsLoading());
+
+          final emailDraftSaved = await _emailRepository.saveEmailAsDrafts(
+            createEmailRequest.session,
+            createEmailRequest.accountId,
+            emailCreated,
+            cancelToken: cancelToken
+          );
+
+          yield dartz.Right<Failure, Success>(
+            SaveEmailAsDraftsSuccess(
+              emailDraftSaved.id!,
+              currentMailboxState: listCurrentState?.value1,
+              currentEmailState: listCurrentState?.value2
+            )
+          );
+        } else {
+          yield dartz.Right<Failure, Success>(UpdatingEmailDrafts());
+
+          final emailDraftSaved = await _emailRepository.updateEmailDrafts(
+            createEmailRequest.session,
+            createEmailRequest.accountId,
+            emailCreated,
+            createEmailRequest.draftsEmailId!,
+            cancelToken: cancelToken
+          );
```

Comment:
why not destroy the old one inside `update`

Reply:
They are two independent events, we need to separate these two processing streams.